pkg/process/checks/process.go

```diff
@@ -78,9 +90,17 @@ func (p *ProcessCheck) RealTime() bool { return false }
 // limit the message size on intake.
 // See agent.proto for the schema of the message and models used.
 func (p *ProcessCheck) Run(cfg *config.AgentConfig, groupID int32) ([]model.MessageBody, error) {
-	p.Lock()
-	defer p.Unlock()
+	results, err := p.run(cfg, groupID, false)
+	if err != nil {
+		return nil, err
+	}
+	if len(results) == 0 {
+		return nil, nil
+	}
+	return results[0].Messages, nil
```

Comment:
Sorry I was a bit confused by this, sounds like the `results` is only going to have either one or two items depend on whether it's RT or not? If so why not creating a struct? I have not reviewed the whole thing so I could be missing something.

Reply:
I agree with you, it's quite clunky as is in the end, I will change results to be a struct with two fields.